CoreScriptsRoot/Modules/Panel3D.lua

```diff
@@ -120,42 +244,69 @@ end
 
 local zeroVector = Vector3.new(0, 0, 0)
 local baseHorizontal = CFrame.new()
+local basePosition = Vector3.new()
 local hitAny = false
```

Comment:
Now with the new updates I think this is a confusing name, we should call it something like currentHoverPanel/Part
and initialized it to nil instead of false